java-checks/src/test/files/checks/ReuseRandomCheck.java

```diff
@@ -0,0 +1,29 @@
+import java.util.Random;
+
+public class A {
```

Comment:
You lack a case where the instantiated object is assigned to a field and/or a static field.
You also lack a case where the left expression of an assignement is not a variable : `this.a`
Moreover : what about such cases  : `a = b = c = new Random();`